prboom2/src/r_bsp.c

```diff
@@ -717,8 +717,13 @@ static void R_Subsector(int num)
          * a future patch by refactoring this into the renderer and tagging
          * visible candidate sectors during drawing.
          */
-        if (frontsector->floorheight >= viewz && (frontsector->flags & MISSING_BOTTOMTEXTURES))
-        {
+        if ((frontsector->floorheight >= viewz ||
+             viewz - frontsector->floorheight >= (500 << FRACBITS)) &&
```

Comment:
Could you add a constant like `FLOOR_BLEED_THRESHOLD` and put the fixme comment above a define for that?